gio/src/subclass/socket_control_message.rs

```diff
@@ -0,0 +1,182 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{subclass::prelude::*, translate::*, Cast};
+
+use crate::SocketControlMessage;
+
+pub trait SocketControlMessageImpl: ObjectImpl + SocketControlMessageImplExt + Send {
+    fn level(&self) -> i32 {
+        self.parent_level()
+    }
+
+    fn msg_type(&self) -> i32 {
+        self.parent_msg_type()
+    }
+
+    fn size(&self) -> usize {
+        self.parent_size()
+    }
```

Comment:
Without implementing `serialize` and `deserialize` this is all rather useless or not?

Reply:
For some cases `serialize` is enough (e.g. to pass a `SO_TXTIME` to a socket) but I agree that a more complete implementation makes more sense.